src/util/mod.rs

```diff
@@ -81,3 +81,60 @@ impl error::Error for HexError {
         }
     }
 }
+
+#[cfg(test)]
+pub mod test {
+    use super::*;
+    use std::sync::mpsc::sync_channel;
+    use std::process;
+
+    pub fn with_timeout<F>(timeout_secs: u64, test_func: F)
+    where
+        F: FnOnce() -> () + std::marker::Send + 'static
+    {
+        let (sx, rx) = sync_channel(1);
+
+        // run test in separate thread, so a main-thread panic ends the test
```

Comment:
Do tests fail if a non-main thread panics?

Reply:
It doesn't right now, but it will if I set a panic hook.

Reply:
Fixed -- if the test panics in an unwindable way (i.e. _not_ through a signal), it will be caught and translated into a test failure.  Similarly, if the test times out, it will panic and exit without causing the whole test framework to abort.